### PR TITLE
new configuration for using nf-core pipelines at University of Luebeck

### PR DESCRIPTION
**Pull request by finjasygo**
---
name: New Config
about: A new cluster config
---

Please follow these steps before submitting your PR:

- [X] If your PR is a work in progress, include `[WIP]` in its title
- [X] Your PR targets the `master` branch
- [X] You've included links to relevant issues, if any

Steps for adding a new config profile:

- [X] Add your custom config file to the `conf/` directory
- [X] Add your documentation file to the `docs/` directory
- [X] Add your custom profile to the `nfcore_custom.config` file in the top-level directory
- [X] Add your custom profile to the `README.md` file in the top-level directory
- [X] Add your profile name to the `profile:` scope in `.github/workflows/main.yml`

### CODE REVIEW

File: conf/uzl_omics.config [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    max_time   = 72.h     
+}
+
+process {
+    executor = 'slurm'
+    queue = 'shortterm'
+    scratch = '$SCRATCH'
+}
+
+workDir = "$WORK/nf-core_rnaseq_workdir"
```
Comment (pontus) [inline conf/uzl_omics.config:19]: Forcing `workdir` in an institutional configuration does not make sense to me?

File: conf/uzl_omics.config [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+singularity {
+    enabled = true
+    runOptions = "-B $WORK/nf-core_rnaseq_workdir -B $HOME/.nextflow/assets/nf-core/rnaseq/bin"
+    cacheDir = "$WORK/.singularity"
```
Comment (pontus) [inline conf/uzl_omics.config:24]: Mixing up the nextflow singularity cache with the directory used for (the different) cache used by singularity is probably not a good idea. Is there a problem with keeping it in the `workDir`?

File: conf/uzl_omics.config [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    executor = 'slurm'
+    queue = 'shortterm'
+    scratch = '$SCRATCH'
+}
+
+workDir = "$WORK/nf-core_rnaseq_workdir"
+
+singularity {
+    enabled = true
+    runOptions = "-B $WORK/nf-core_rnaseq_workdir -B $HOME/.nextflow/assets/nf-core/rnaseq/bin"
```
Comment (pontus) [inline conf/uzl_omics.config:23]: These are very specific for an institutional configuration.

My recommendation (for singularity) is to have the configuration bind all "project" file spaces by default, but for these, do you need more than `autoMounts = true`?

File: conf/uzl_omics.config (hunk 1 above)
Comment (jfy133) [reply conf/uzl_omics.config:19]: Agreed, we have pipeline specific institutional comfigs for this! Check `conf/pipelines`

**Review by finjasygo — COMMENTED** [review]
Updated the files according to your suggestions.

File: docs/uzl_omics.md [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+The rnaseq nf-core pipeline has been successfully configured for use on the UzL OMICS cluster at the University of Luebeck.
+Implementation of nf-core pipelines for use on the UzL OMICS is in process. 
+
+To use, run the pipeline with `-profile uzl_omics`. This will download and launch the `uzl_omics.config` which has been pre-configured with a setup suitable for the UzL OMICS cluster. Using this profile, a docker image containing all of the required software will be downloaded, and converted to a Singularity image before execution of the pipeline.
+
+Before running the pipeline you will need to load Nextflow and Singularity using the environment module system on UzL OMICS cluster. You can do this by issuing the commands below:
+
+```bash
+module load nextflow
+module load singularity
```
Comment (pontus) [inline docs/uzl_omics.md:12]: Are loaded modules carried through to jobs or should this be in a `process.beforeScript` as well?

File: docs/uzl_omics.md [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+```
+
+nf-core requires a Nextflow version 22.10.1 or higher, so you have to intall a more recent version first.
+
+For Nextflow versions newer than 23.07.0-edge, it is necessary to explicitly mount the home directory using the command:
```
Comment (pontus) [inline docs/uzl_omics.md:17]: There seems to be some buggy modules not being happy without a writable `$HOME`, is this because of this or some other thing?

File: docs/uzl_omics.md [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+
+```bash
+NXF_SINGULARITY_HOME_MOUNT=true
+```
+
+You need to run the pipeline in `$WORK`. You can specify this in your batch-script, as in the following expample:
```
Comment (pontus) [inline docs/uzl_omics.md:23]: Is this really the case or an old remnant?

**Comment by pontus** [comment]
Looks much better, but I made a few new comments on the documentation. 

Also, the linter still complains on the documentation page, can you please update? (Can e.g. be done with running `prettier` through docker if you don't want to install it on some system).

File: docs/uzl_omics.md (hunk 4 above)
Comment (finjasygo) [reply docs/uzl_omics.md:12]: The modules are carried through. It's not needed in a `process.beforeScript`

File: docs/uzl_omics.md (hunk 6 above)
Comment (finjasygo) [reply docs/uzl_omics.md]: You're right. In previous tests it failed that the pipeline didn't run in `$WORK`. But after testing it again, it works now. I erased it.

**Review by finjasygo — COMMENTED** [review]
Fix according to the suggestions. Updated the documentation after running `prettier`.

File: docs/uzl_omics.md (hunk 5 above)
Comment (pontus) [reply docs/uzl_omics.md:17]: It seems I made the last comments there ;)

Was the outcome that the singularity setup was intentionally broken (not allowing mount controls)? I guess that would limit to `$HOME` if that was the case, but then again, I'd definitely try raising it for a chance to have it fixed first.

Comment (finjasygo) [reply docs/uzl_omics.md:17]: Could you please explain what you mean and how we can get it fixed?

The singularity is installed by the HPC maintainers, and we are currently unsure why this problem occurs when we do not explicitly ask to mount `$HOME` using the global variable `NXF_SINGULARITY_HOME_MOUNT`.

In our understanding, it is rather a Nextflow related issue, because previous versions (e.g. 22.10.1) of Nextflow worked fine without this variable decoration.  Starting from version 23.07.0-edge, nextflow documentation says that the default is set to false when it comes to mounting the home directory.

Comment (pontus) [reply docs/uzl_omics.md:17]: As was mentioned in the slack thread, it is very likely a good idea to reach out to your systems people and let them now that their singularity configuration (disallowing user binds) makes things more difficulty for you.

I obviously don't know how your systems are set up exactly, and maybe everything is under users' home directories - then making sure nextflow doesn't disable home binds would be good enough (although the user experience is lousy as it currently requires setting that environment variable). But obviously that doesn't cover anything run outside of the home directories.

A setup that doesn't have singularity stopping if a bind is provided could use `singularity.autoMounts` to pass the binds nextflow knows of that would be required (although in some cases with links going back and forth, it might be a good idea to help with specifying some binds explicitly).

So, to sum up, I'd say the singularity configuration is weird if it doesn't allow the user specificying bind mounts (which it seemed from the slack thread), and I'd generally suggest trying to fix that since it breaks a lot of use cases.

Comment (finjasygo) [reply docs/uzl_omics.md]: We will work on fixing this issue. For now I have edited the documentation file and erased this part, to not cause any conflicts.

**Review by finjasygo — COMMENTED** [review]
edited the documentation file

**Review by pontus — APPROVED** [review]
